packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1634,16 +1637,29 @@ Run 'flutter -h' (or 'flutter <command> -h') for available flutter commands and
   ///
   /// If no flag named [name] was added to the [ArgParser], an [ArgumentError]
   /// will be thrown.
-  bool boolArg(String name) => argResults![name] as bool;
+  bool boolArg(String name, {bool global = false}) {
+    if (global) {
+      return globalResults![name] as bool;
+    }
+    return argResults![name] as bool;
+  }
 
   /// Gets the parsed command-line option named [name] as a `String`.
   ///
   /// If no option named [name] was added to the [ArgParser], an [ArgumentError]
   /// will be thrown.
-  String? stringArg(String name) => argResults![name] as String?;
+  String? stringArg(String name, {bool global = false}) {
+    if (global) {
+      return globalResults![name] as String?;
+    }
+    return argResults![name] as String?;
+  }
 
   /// Gets the parsed command-line option named [name] as `List<String>`.
-  List<String> stringsArg(String name) {
+  List<String> stringsArg(String name, {bool global = false}) {
+    if (global) {
+      return globalResults![name]! as List<String>? ?? <String>[];
```

Comment:
I know you were copying the existing code, but the existing code looks wrong: if I'm reading this right, the left hand side of the expression can never null because of the second `!`, and thus we would never hit the fallback literal.

Reply:
You're right, in fact, this value should never be null.
https://github.com/dart-lang/args/blob/a9543c021f9409832b1668f9256f247585362389/lib/src/option.dart#L149-L153